darts/timeseries.py

```diff
@@ -140,156 +582,251 @@ def pd_dataframe(self, copy=True) -> pd.DataFrame:
+
+        Parameters
+        ----------
+        quantile
+            The desired quantile value. The value must be represented as a fraction
+            (between 0 and 1 inclusive). For instance, `0.5` will return a TimeSeries
+            containing the median of the (marginal) distribution of each component.
 
-    def start_time(self) -> pd.Timestamp:
+        Returns
+        -------
+        TimeSeries
+            The TimeSeries containing the desired quantile for each component.
         """
+        return TimeSeries.from_dataframe(self.quantile_df(quantile))
+
+    def quantiles_df(self, quantiles: Tuple[float] = (0.1, 0.5, 0.9)) -> pd.DataFrame:
+        """
+        Returns a Pandas DataFrame containing the desired quantiles of each component (over the samples).
+        Each of the series components will appear as a column in the DataFrame. The column will be named
+        "<component>_X", where "<component>" is the column name corresponding to this component, and "X"
+        is the quantile value.
+        The quantiles represent the marginal distributions of the components of this series.
+
+        This works only on stochastic series (i.e., with more than 1 sample)
+
+        Parameters
+        ----------
+        quantiles
+            Tuple containing the desired quantiles. The values must be represented as fractions
+            (between 0 and 1 inclusive). For instance, `(0.1, 0.5, 0.9)` will return a DataFrame
+            containing the 10th-percentile, median and 90th-percentile of the (marginal) distribution of each component.
+
         Returns
         -------
-        pandas.Timestamp
-            A timestamp containing the first time of the TimeSeries.
+        pandas.DataFrame
+            The Pandas DataFrame containing the quantiles for each component.
         """
-        return self._df.index[0]
+        # TODO: there might be a slightly more efficient way to do it for several quantiles at once with xarray...
+        return pd.concat([self.quantile_df(quantile) for quantile in quantiles], axis=1)
 
-    def end_time(self) -> pd.Timestamp:
+    def start_time(self) -> Union[pd.Timestamp, int]:
         """
         Returns
         -------
-        pandas.Timestamp
-            A timestamp containing the last time of the TimeSeries.
+        Union[pandas.Timestamp, int]
+            A timestamp containing the first time of the TimeSeries (if indexed by DatetimeIndex),
+            or an integer (if indexed by RangeIndex)
         """
-        return self._df.index[-1]
+        return self._time_index[0]
+
+    def end_time(self) -> Union[pd.Timestamp, int]:
+        """
+        Returns
+        -------
+        Union[pandas.Timestamp, int]
+            A timestamp containing the last time of the TimeSeries (if indexed by DatetimeIndex),
+            or an integer (if indexed by RangeIndex)
+        """
+        return self._time_index[-1]
 
     def first_value(self) -> float:
```

Comment:
I don't really see an application for this function, but if it stays, could we also make it a property method?
Otherwise I'd rather have a method called head(n=5) which covers multivariate and univariate time series at the same time. This differentiation here doesn't really seem necessary.

Reply:
You're right, agree. However I'd rather leave this for later.